src/physics/initial_conditions/initial_condition_function.h

```diff
@@ -335,6 +335,99 @@ class InitialConditionFunction_KHI : public InitialConditionFunction<dim,nstate,
 
 };
 
+/// Initial Condition Function: 1D Sod Shock Tube
+/** See Chen & Shu, Entropy stable high order discontinuous 
+*   Galerkin methods with suitable quadrature rules for hyperbolic 
+*   conservation laws., 2017, Pg. 25
+*/
+template <int dim, int nstate, typename real>
+class InitialConditionFunction_SodShockTube: public InitialConditionFunction<dim,nstate,real>
+{
+protected:
+    using dealii::Function<dim,real>::value; ///< dealii::Function we are templating on
+
+    // Euler physics pointer. Used to convert primitive to conservative.
+    std::shared_ptr < Physics::Euler<dim, nstate, double > > euler_physics;
+
+public:
+    /// Constructor for InitialConditionFunction_SodShockTube
+    /** Calls the Function(const unsigned int n_components) constructor in deal.II*/
+    InitialConditionFunction_SodShockTube (
+            Parameters::AllParameters const* const param);
+
+    /// Value of initial condition
+    real value (const dealii::Point<dim,real> &point, const unsigned int istate = 0) const override;
+};
+
+/// Initial Condition Function: 2D Low Density Euler
+/** See Zhang & Shu, On positivity-preserving high order 
+*   discontinuous Galerkin schemes for compressible Euler 
+*   equations on rectangular meshes, 2010 Pg. 10
+*/
+template <int dim, int nstate, typename real>
+class InitialConditionFunction_LowDensity2D: public InitialConditionFunction<dim,nstate,real>
+{
+protected:
+    using dealii::Function<dim,real>::value; ///< dealii::Function we are templating on
+
+    // Euler physics pointer. Used to convert primitive to conservative.
+    std::shared_ptr < Physics::Euler<dim, nstate, double > > euler_physics;
+public:
+    /// Constructor for InitialConditionFunction_LowDensity2D
+    /** Calls the Function(const unsigned int n_components) constructor in deal.II*/
+    InitialConditionFunction_LowDensity2D (
+            Parameters::AllParameters const* const param);
+
+    /// Value of initial condition
+    real value (const dealii::Point<dim,real> &point, const unsigned int istate = 0) const override;
+};
+
+/// Initial Condition Function: 1D Leblanc Shock Tube
+/** See Zhang & Shu, On positivity-preserving high order
+*   discontinuous Galerkin schemes for compressible Euler
+*   equations on rectangular meshes, 2010 Pg. 14
+*/
+template <int dim, int nstate, typename real>
+class InitialConditionFunction_LeblancShockTube: public InitialConditionFunction<dim,nstate,real>
+{
+protected:
+    using dealii::Function<dim,real>::value; ///< dealii::Function we are templating on
+
+    // Euler physics pointer. Used to convert primitive to conservative.
+    std::shared_ptr < Physics::Euler<dim, nstate, double > > euler_physics;
```

Comment:
use a `unique_ptr` if you can

Reply:
I tried changing it to unique_ptr but the build kept failing. I think it is because physics_factory returns a shared_ptr for Euler. Should I change physics_factory so I can use a unique_ptr here or should I just leave it as a shared_ptr?

Reply:
Definitely don't change the factory, since you know that this initial condition will always depend on an euler physics, you can simply manually create the euler_physics object by calling the constructor yourself; see what I did here for `InitialConditionFunction_TurbulentChannelFlow` class; 
- header file definition of the class:
https://github.com/dougshidong/PHiLiP/blob/99b46e047d9552ba3f759567275685588bdeb120/src/physics/initial_conditions/initial_condition_function.h#L75
- where the object gets created:
https://github.com/dougshidong/PHiLiP/blob/99b46e047d9552ba3f759567275685588bdeb120/src/physics/initial_conditions/initial_condition_function.cpp#L694-L727

Reply:
I just saw what you did with the IC Euler Base class, that looks great. You can resolve this comment